src/librustc_mir/borrow_check/mod.rs

```diff
@@ -2693,7 +2678,7 @@ impl<'cx, 'gcx, 'tcx> MirBorrowckCtxt<'cx, 'gcx, 'tcx> {
 }
 
 #[derive(Copy, Clone, PartialEq, Eq, Debug)]
-struct Context {
+pub struct Context {
```

Comment:
this probably does not need to be `pub`...? in particular, submodules can still access it when it is priv

Reply:
```
error[E0446]: private type `borrow_check::Context` in public interface
  --> src/librustc_mir/borrow_check/nll/explain_borrow/mod.rs:21:5
   |
21 | /     pub fn explain_why_borrow_contains_point(
22 | |         &self,
23 | |         context: Context,
24 | |         borrow: &BorrowData<'_>,
...  |
78 | |         }
79 | |     }
   | |_____^ can't leak private type

error: aborting due to previous error
```

Reply:
change that `pub fn explain` to `pub(super) fn explain`

Reply:
:+1: